src/components/MakeWidget/subcomponents/ExpirySelector/ExpirySelector.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState, useEffect } from "react";
+
+import { Dropdown, SelectOption } from "../../../Dropdown/Dropdown";
+import { Wrapper, Title, Input } from "./ExpirySelector.styles";
+
+const MS_PER_MINUTE = 60000;
+
+const options = [
+  {
+    label: "minutes",
+    value: `${MS_PER_MINUTE}`,
+  },
+  {
+    label: "hours",
+    value: `${60 * MS_PER_MINUTE}`,
+  },
+  {
+    label: "days",
+    value: `${24 * 60 * MS_PER_MINUTE}`,
+  },
+  {
+    label: "weeks",
+    value: `${7 * 24 * 60 * MS_PER_MINUTE}`,
+  },
+];
+
+const floatRegExp = new RegExp("^([0-9])*$");
+
+export type ExpirySelectorProps = {
+  updateExpiry: (msToExpiry: number) => void;
+};
+
+export const ExpirySelector: React.FC<ExpirySelectorProps> = ({
+  updateExpiry,
+}) => {
+  const [unit, setUnit] = useState(options[0]);
+  const [amount, setAmount] = useState("1");
+
+  useEffect(() => {
+    updateExpiry((parseInt(amount) || 1) * parseInt(unit.value));
+  }, [unit, amount]);
+
+  function handleUnitChange(option: SelectOption) {
+    setUnit(option);
+  }
+
+  function handleAmountChange(value: string) {
+    if (value === "" || floatRegExp.test(value)) {
+      value = value.replace(/^0+/, "0");
+      setAmount(value);
+    }
+  }
+
+  return (
+    <Wrapper>
+      <Title>EXPIRES IN</Title>
```

Comment:
Add translation for this. You can add it temporarily in locales/en/translation.json. Did you get setup yet at poeditor? That's where we manage all translations.

Reply:
not set up in poeditor. ill ping don.